src/corppa/poetry_detection/passim/get_passim_excerpts.py

```diff
@@ -0,0 +1,82 @@
+"""
+Gather excerpts from passim page-level results.
+"""
+
+import argparse
+import csv
+import pathlib
+import sys
+
+import orjsonl
+
+
+def get_passage_matches(input_file):
+    """
+    Extracts and yields the passim-identified passage-level matches
+    """
+    for page in orjsonl.stream(input_file):
+        if not page["n_spans"]:
+            # Skipe pages without matches
+            continue
+        for poem_span in page["poem_spans"]:
+            match = poem_span.copy()
+            match["page_id"] = page["page_id"]
+            yield match
+
+
+def save_passim_passage_matches(input_file, output_file):
+    with open(output_file, mode="w", newline="") as tsvfile:
+        fieldnames = [
+            "page_id",
+            "ref_id",
+            "ref_corpus",
+            "page_start",
+            "page_end",
+            "ref_start",
+            "ref_end",
+            "page_excerpt",
+            "ref_excerpt",
+            "aligned_page_excerpt",
+            "aligned_ref_excerpt",
```

Comment:
what are the differences= between the excerpts and the aligned excerpts? in most cases as I'm scanning they seem exactly the same - is one of them the version that includes the passim markings for extra/missing characters?

Reply:
The aligned excerpts are the character-level alignment output used by passim. They are never exactly the same "-" are used any time an insertion/removal happens in the alignment algorithm.

Reply:
Passim does not return the original page / reference text excerpts; only the "aligned" versions. That's why the external text corpora need to be provided because it's not included in the passim results files themselves.